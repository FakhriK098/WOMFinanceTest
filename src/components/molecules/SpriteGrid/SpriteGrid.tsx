import React from 'react'
import { Image, StyleSheet, View } from 'react-native'
import { spacing } from '@theme/spacing'

type Props = { sprites: string[] }

const SpriteGrid = ({ sprites }: Props) => {
  return (
    <View style={styles.grid}>
      {sprites.map((s, i) => (
        <Image key={`${s}-${i}`} source={{ uri: s }} style={styles.img} resizeMode="contain" />
      ))}
    </View>
  )
}

const styles = StyleSheet.create({
  grid: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: spacing.sm,
  },
  img: {
    width: 72,
    height: 72,
  },
})

export default React.memo(SpriteGrid)
