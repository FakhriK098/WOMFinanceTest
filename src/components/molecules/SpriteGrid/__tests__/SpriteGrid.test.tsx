import React from 'react';
import renderer, { act } from 'react-test-renderer';

import SpriteGrid from '../SpriteGrid';

describe('SpriteGrid molecule', () => {
  it('matches snapshot and renders images', () => {
    const sprites = ['http://a', 'http://b', 'http://c'];
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(<SpriteGrid sprites={sprites} />);
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    const images = tr!.root.findAll((inst) => inst.type === 'Image');
    expect(images.length).toBe(3);
    act(() => tr!.unmount());
  });
});

