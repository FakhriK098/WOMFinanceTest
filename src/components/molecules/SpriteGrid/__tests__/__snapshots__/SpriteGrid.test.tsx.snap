// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpriteGrid molecule matches snapshot and renders images 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": 8,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "http://a",
      }
    }
    style={
      {
        "height": 72,
        "width": 72,
      }
    }
  />
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "http://b",
      }
    }
    style={
      {
        "height": 72,
        "width": 72,
      }
    }
  />
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "http://c",
      }
    }
    style={
      {
        "height": 72,
        "width": 72,
      }
    }
  />
</View>
`;
